out/notebooks/spl2/installer.ts

```diff
@@ -0,0 +1,514 @@
+import axios from 'axios';
+import * as child_process from 'child_process';
+import { XMLParser} from 'fast-xml-parser';
+import * as fs from 'fs';
+import * as path from 'path';
+import { pipeline } from 'stream';
+import * as tar from 'tar-fs';
+import * as unzipper from 'unzipper';
+import * as util from 'util';
+import { env, ExtensionContext, StatusBarItem, Uri, window, workspace } from 'vscode';
+import * as zlib from 'zlib';
+
+// Keys used to store/retrieve state related to this extension
+export const configKeyAcceptedTerms = 'splunk.spl2.acceptedTerms';
+export const configKeyJavaPath = 'splunk.spl2.javaPath';
+export const configKeyLspVersion = 'splunk.spl2.languageServerVersion';
+
+export const stateKeyLatestLspVersion = 'splunk.spl2.latestLspVersion';
+export const stateKeyLastLspCheck = 'splunk.spl2.lastLspCheck';
+
+// Minimum version of Java needed for SPL2 Language Server
+const minimumMajorJavaVersion = 17;
+
+export enum TermsAcceptanceStatus {
+    DeclinedForever = 'declined (forever)',
+    DeclinedOnce = 'declined (once)',
+    Accepted = 'accepted',
+}
+
+/**
+ * Provide a guided install experience for installing Java and SPL2 Language Server including
+ * accepting Splunk General terms. If compatible Java and Language Server is already installed
+ * this will be a no-op.
+ */
+export async function getMissingSpl2Requirements(context: ExtensionContext, progressBar: StatusBarItem): Promise<boolean> {
```

Comment:
getMissingSpl2Requirements indicates that it will return a list of missing requirements. Maybe rename to `installMissingSpl2Requirements`?